hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenCreateSpecs.java

```diff
@@ -279,17 +282,18 @@ public HapiSpec autoRenewValidationWorks() {
                         cryptoCreate(deletingAccount).balance(0L))
                 .when(
                         cryptoDelete(deletingAccount),
-                        tokenCreate(PRIMARY)
-                                .autoRenewAccount(deletingAccount)
-                                .hasKnownStatus(INVALID_AUTORENEW_ACCOUNT),
+                        tokenCreate(PRIMARY).autoRenewAccount(deletingAccount).hasKnownStatus(ACCOUNT_DELETED),
```

Comment:
We have a `TokenHandlerHelper.getIfUsable` method - this is a general method that checks if the account was deleted - if so we return the general error code _ACCOUNT_DELETED_(without checking for additional details and return specific response codes). This fails in `creationValidatesTreasuryAccount`, `validateNewTokenAssociations`, and `autoRenewValidationWorks` where we expect a concrete deleted response(_INVALID_TREASURY_ACCOUNT_FOR_TOKEN_ and _INVALID_AUTORENEW_ACCOUNT_). For example, _INVALID_TREASURY_ACCOUNT_FOR_TOKEN_, when the account is deleted **AND** the account is treasury -  we return _INVALID_TREASURY_ACCOUNT_FOR_TOKEN_. Do we need to add the specific checks or is it okay to return the general _ACCOUNT_DELETED_ in these cases?

Reply:
I assume this change will break the tests for the current code base. If we agree to change the expectation, we must preserve the old expectation.

A more specific error message is preferred IMO. A new token can be created with several keys, and knowing which one is invalid is helpful.

Reply:
We should preserve mono-service behavior. In this case if we need specific error codes, we can add a new method with different signature of `getIfUsable` and provide the error response to be returned.